### DISCUSSION TITLE
session id cookies are not cleared

### DISCUSSION THREAD

**Discussion by crossit** [adonisjs/core#3528] [Help]
Hey everyone,

we're using @adonisjs/session with the following config

```
driver: 'cookie',
cookieName: 'adonis-session',
  clearWithBrowser: true,
  age: '2h',
  cookie: {
    path: '/',
    httpOnly: true,
    sameSite: false,
  },
```

and accessing a seperate view

`Route.on('/shop').render('shop')`

If I remove or rename the adonis-session Cookie, a new crypt-named cookie appears. Same behavior when I'm entering /shop after 2  hours.
So we (and normal visitors) keep getting new crypt-named cookies and at some point the header fields size of the server exceeds. Same with  clearWithBrowser: false

<img width="995" alt="Bildschirmfoto 2022-02-01 um 11 04 04" src="https://user-images.githubusercontent.com/16023907/151949588-118ce1cf-4e4b-4923-a9c4-6dd625991b6f.png">

Any idea what the issue could be?

Thanks!

**Comment by thetutlage** [comment]
> If I remove or rename the adonis-session Cookie, a new crypt-named cookie appears

Can you share how you remove the `adonis-session` cookie?

**Reply by crossit** [reply]
I just remove it from the dev console, reload the page, new adonis-session cookie appears plus a new crypt-named cookie.
same behavior when I'm reloading the page after 2h.

**Reply by thetutlage** [reply]
Yup. Coz the `adonis-session` holds the session id and that cryptic cookie is an hash of that session id and holds the session value inside it. So when you manually delete the `adonis-session`, the backend creates a new session id + new hash holding value appears.

**Reply by crossit** [reply]
Thanks for your answer! Okay so its expected behavior. But what about the normal visitors who access the site from time to time and keep getting new cryptic cookies?

**Reply by thetutlage** [reply]
Shouldn't happen with normal visitors. Since, both the `adonis-session` and the cryptic cookie expires together.
